File: leetCodeSolutions/src/twoSum.cpp
// Found in https://leetcode.com/problems/two-sum/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        unordered_map<int,int> haveSeenIt;
        for(int i = 0; i < nums.size(); i++) {
            if(haveSeenIt.find(target - nums[i]) != haveSeenIt.end()) {
                res.push_back(haveSeenIt[target - nums[i]]);
                res.push_back(i);
                return res;
            }
            haveSeenIt.insert(pair<int, int>(nums[i], i));
        }
        return res;
    }
};

File: leetCodeSolutionsPython/twoSum.py
# Found in https://leetcode.com/problems/two-sum/
#
# Time Complexity:     O(n)
# Space Complexity:    O(n)

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """(Y

        haveSeenIt = {}
        for idx, val in enumerate(nums):
            if (target - val) in haveSeenIt:
                return [haveSeenIt[target - val], idx]
            haveSeenIt[val] = idx
        return []

